### ISSUE TITLE
Addon "done" event not called in 3.X

### ISSUE THREAD

**Issue by Caligatio** [mitmproxy/mitmproxy#3206]
##### Steps to reproduce the problem:

1. Create an addon using the "done" event
2. Start mitmdump or mitmproxy with your addon specified using -s
3. Exit mitmproxy/mitmdump

##### Any other comments? What have you tried so far?

Here's a test addon that illustrates the problem:
```python
class TestAddon():
    def done(self):
        print('Hello, World!')

addons = [
    TestAddon()
]
```
I am forced to use mitmproxy 3.X as we still use Ubuntu Xenial extensively across our infrastructure which uses Python 3.5.  The below system dump reflects a VM I spun up to test this in isolation.  This problem has been fixed in 4.X but it would be really great if it could be backported to 3.X as this is a killer bug for me (my add-on finalizes an output file in "done" and worked great in 2.X).

##### System information
Mitmproxy: 3.0.4 
Python:    3.6.5
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Linux-4.15.0-23-generic-x86_64-with-Ubuntu-18.04-bionic

**Comment by mhils** [comment]
Thanks. It looks like your mitmproxy is outdated. Could you please upgrade an try again?

**Comment by Caligatio** [comment]
As a stated in the original issue, I cannot upgrade to the 4.X line as it has a hard dependency on Python 3.6.  Is there is a newer version available that still supports Python <3.6?

**Comment by mhils** [comment]
I'm afraid we're not going to ship any patches for 3.x, sorry. Things you can do:

- Use the precompiled binaries from https://mitmproxy.org/downloads/ (that works unless you need additional Python packages in your scripts)
- Install Python 3.6 on Xenial, e.g. via the deadsnakes PPA
- Docker
- Use mitmproxy 2 (unsupported from our end)
